twopaneview/examples/TwoPaneView.js

```diff
@@ -0,0 +1,215 @@
+import React, {Fragment, Component} from 'react';
+import {
+  SafeAreaView,
+  StyleSheet,
+  ScrollView,
+  View,
+  ViewProps,
+  Text,
+  StatusBar,
+  Dimensions,
+} from 'react-native';
+import { DualScreenInfo } from 'react-native-dualscreen'
+
+const TPV_Orientation = {
+  Horizontal: 'horizontal',
+  Vertical: 'vertical',
+};
+
+const TPV_PanePriority = {
+  Pane1: 'pane1',
+  Pane2: 'pane2',
+};
+
+const PaneMode = {
+  Auto: 'Auto',
+  Single: 'Single',
+  Dual: 'Dual',
+};
+
+export default class TwoPaneView extends Component {
+  state = {
+    dims: Dimensions.get('window'),
+    spanning: DualScreenInfo.isSpanning, 
+    panePriority: this.props.panePriority,
+    panePriorityVerticalSpanning: this.props.panePriorityVerticalSpanning,
+    paneMode: this.props.paneMode ? this.props.paneMode: PaneMode.Auto,
+    twoPaneWidth: this.props.twoPaneWidth ? this.props.twoPaneWidth: 640,
+    onModeChanged: this.props.onModeChanged ? this.props.onModeChanged : ()=>{},
+  };
+
+  componentDidMount() {
+    Dimensions.addEventListener('change', this._handleDimensionsChange);
+  }
+
+  componentWillUnmount() {
+    Dimensions.removeEventListener('change', this._handleDimensionsChange);
+  }
+
+  _handleDimensionsChange = dimensions => {
```

Comment:
you dont need it since DualScreenInfo updating isSpanning automatically and you need only to subscribe there